### ISSUE TITLE
Exporting ONNX OBB models But the dimensions of the output results are inconsistent

### ISSUE THREAD

**Issue by hacktmz** [ultralytics/ultralytics#11741]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Exporting ONNX OBB models But the dimensions of the output results are inconsistent
`yolo export model=yolov8s-obb.pt format=onnx opset=15`

The original model output dimension is [1,20,21504]
`PyTorch: starting from 'yolov8s-obb.pt' with input shape (1, 3, 1024, 1024) BCHW and output shape(s) (1, 20, 21504) (22.2 MB)`

But using the model I trained with imgs=640 parameters and exporting the onnx model using the following command, the output changed.

`yolo obb train data=obb.yaml model=yolov8s-obb.pt epochs=1000 imgsz=640 batch=64 patience=20 workers=8 single_cls=True`

yolo export model=runs/obb/train17/weights/best.pt format=onnx opset=15
Ultralytics YOLOv8.2.10 🚀 Python-3.8.10 torch-1.14.0a0+44dac51 CPU (Intel Xeon Platinum 8255C 2.50GHz)
PyTorch: starting from 'runs/obb/train17/weights/best.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 6, 8400) (44.4 MB)
ONNX: starting export with onnx 1.12.0 opset 15...
========== Diagnostic Run torch.onnx.export version 1.14.0a0+44dac51 ===========

But after replacing imgsz=1024 to train, the exported result is the correct dimension [1,6,21504]

### Environment

Ultralytics YOLOv8.2.10 🚀 Python-3.8.10 torch-1.14.0a0+44dac51 CUDA:0 (Tesla V100-SXM2-32GB, 32510MiB)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by hacktmz** [comment]
root@zhenty-33-0:/gpu2-data/zhenty/yolov8# yolo export model=runs/obb/train/weights/best.pt format=onnx opset=15  imgsz=640
Ultralytics YOLOv8.2.10 🚀 Python-3.8.10 torch-1.14.0a0+44dac51 CPU (Intel Xeon Platinum 8255C 2.50GHz)
YOLOv8s-obb summary (fused): 187 layers, 11411958 parameters, 0 gradients, 29.4 GFLOPs

PyTorch: starting from 'runs/obb/train/weights/best.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 6, 8400) (44.9 MB)

ONNX: starting export with onnx 1.12.0 opset 15...
========== Diagnostic Run torch.onnx.export version 1.14.0a0+44dac51 ===========
verbose: False, log level: Level.ERROR
======================= 0 NONE 0 NOTE 0 WARNING 0 ERROR ========================

ONNX: export success ✅ 2.7s, saved as 'runs/obb/train/weights/best.onnx' (43.7 MB)

Export complete (6.4s)
Results saved to /gpu2-data/zhenty/yolov8/runs/obb/train/weights
Predict:         yolo predict task=obb model=runs/obb/train/weights/best.onnx imgsz=640  
Validate:        yolo val task=obb model=runs/obb/train/weights/best.onnx imgsz=640 data=/gpu2-data/dataset/ocrdata20240430/obb.yaml  
Visualize:       https://netron.app
💡 Learn more at https://docs.ultralytics.com/modes/export
root@zhenty-33-0:/gpu2-data/zhenty/yolov8# yolo export model=runs/obb/train/weights/best.pt format=onnx opset=15  imgsz=1024
Ultralytics YOLOv8.2.10 🚀 Python-3.8.10 torch-1.14.0a0+44dac51 CPU (Intel Xeon Platinum 8255C 2.50GHz)
YOLOv8s-obb summary (fused): 187 layers, 11411958 parameters, 0 gradients, 29.4 GFLOPs

PyTorch: starting from 'runs/obb/train/weights/best.pt' with input shape (1, 3, 1024, 1024) BCHW and output shape(s) (1, 6, 21504) (44.9 MB)

ONNX: starting export with onnx 1.12.0 opset 15...
========== Diagnostic Run torch.onnx.export version 1.14.0a0+44dac51 ===========
verbose: False, log level: Level.ERROR
======================= 0 NONE 0 NOTE 0 WARNING 0 ERROR ========================

ONNX: export success ✅ 2.9s, saved as 'runs/obb/train/weights/best.onnx' (43.8 MB)

Export complete (8.2s)
Results saved to /gpu2-data/zhenty/yolov8/runs/obb/train/weights
Predict:         yolo predict task=obb model=runs/obb/train/weights/best.onnx imgsz=1024  
Validate:        yolo val task=obb model=runs/obb/train/weights/best.onnx imgsz=1024 data=/gpu2-data/dataset/ocrdata20240430/obb.yaml  
Visualize:       https://netron.app
💡 Learn more at https://docs.ultralytics.com/modes/export
root@zhenty-33-0:/gpu2-data/zhenty/yolov8#

**Comment by glenn-jocher** [comment]
It looks like the output dimensions differ based on the `imgsz` used during the model export process. When the model is trained with `imgsz=640` and exported with the same, the dimensions are adjusted accordingly which differs from when exported with `imgsz=1024`.

It's important to maintain consistency between the training and exporting image sizes to ensure the output dimensions align with expectations. If you need the output dimensions to match a specific size, consider retraining the model with that exact `imgsz` or adjust the `imgsz` during export to match your trained model settings.

If you have further issues or questions, feel free to reach out! 🚀
